content/docs/user-guide/programming/components/reflection/behavior-context.md

```diff
@@ -0,0 +1,461 @@
+    ->Attribute(AZ::Script::Attributes::Module, "atom");
+```
+
+### `AZ_ENUM` utility macro
+
+O3DE `AzCore` provides a utility macro called `AZ_ENUM_DEFINE_REFLECT_UTILITIES` that works in conjunction with the `AZ_ENUM` macros to generate utility functions that can reflect _all_ values of an enum for you, saving you from the need to reflect each value individually. This might be useful for enums that are expected to change during the course of development, or that contain a large number of values. It requires the enum to be defined using either of the AZ_ENUM macros, including `AZ_ENUM`, `AZ_ENUM_WITH_UNDERLYING_TYPE`, `AZ_ENUM_CLASS`, or `AZ_ENUM_CLASS_WITH_UNDERLYING_TYPE`.
+
+To use this macro, include `<AzCore/Preprocessor/EnumReflectUtils.h>` in the source where you wish to reflect the enum.
+
+```cpp
+// Define the enum and its values.
+AZ_ENUM_CLASS(TestEnum, Walk, Run, Fly);
+
+// Generate the utility functions for reflection.
+AZ_ENUM_DEFINE_REFLECT_UTILITIES(TestEnum)
+
+// Call the generated reflect function from within your component's Reflect function.
+TestEnumReflect(*behaviorContext);
+```
+
+For additional context on how to use these macros, examine the test code in `Code\Framework\AzCore\Tests\EnumTests.cpp` or a Gem where they are used, such as `Gems\Atom\RHI\Code\Source\RHI.Reflect\RenderStates.cpp` and `Gems\Atom\RHI\Code\Include\Atom\RHI.Reflect\RenderStates.h`.
+
+## EBus
+
+Enable scripts to send and receive events by binding EBus events and event handlers to the behavior context:
+
+### Events
+
+EBus provides a mechanism to broadcast an event to all handlers or send an event directly to handlers connected at a specific ID. When you bind an event, O3DE automatically reflects `Broadcast`, `Event`, `QueueBroadcast`, and `QueueEvent`, as needed for your EBus configuration.
+
+```cpp
+behaviorContext.EBus<TestBus>("TestBus")->
+    Handler<TestBusHandler>()->
+    Event("SetSum1", &TestBus::Events::SetSum1)->
+    Event("GetSum1", &TestBus::Events::GetSum1)->
+;
+```
+
+Once compiled, the events can be used in Script Canvas and called in Lua:
+
+```lua
+local result = TestBus.Broadcast.GetSum1(1)
+```
+
+### Event handlers
+
+An event handler reflects a class that you must implement to forward messages from the EBus to behavior context methods. You must create a class that can monitor the specified EBus and forward messages to the behavior context. This is a requirement because the behavior context cannot guarantee that there is a handler for each message. If a message expects a result, you must provide a default result in case the message is not handled by the behavior context user. Keep in mind that the system creates as many of these handlers as the behavior context requires. Handlers can also execute in different threads. As a result, you should avoid static storage for values that change. The best way to understand this is to examine the example that follows.
```

Comment:
```suggestion
An event handler reflects a class that you must implement to forward messages from the EBus to behavior context methods. To implement the event handler, you must create a class that can monitor the specified EBus and forward messages to the behavior context.
 
This is required because the behavior context cannot guarantee that there is a handler for each message. If a message expects a result, you must provide a default result in case the behavior context user does not handle the message. Keep in mind that the system creates as many of these handlers as the behavior context requires. Handlers can also execute in different threads. As a result, you should avoid static storage for values that change. For example:
```
* For clarity, context, added "To implement the event handler..." in the second sentence. Verify for accuracy.
* Consider breaking up this long paragraph for easier scannability.
* Minor passive voice fix ("the message is not handled by the...user" to "the...user does not handle the message")
* I recommend avoid claiming that reviewing the following example is "The best way to understand this". You can probably just lead in with "For example:".